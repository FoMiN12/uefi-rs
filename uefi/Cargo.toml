[package]
name = "uefi"
version = "0.28.0"
readme = "README.md"
description = "Safe and easy-to-use wrapper for building UEFI apps."

authors.workspace = true
categories.workspace = true
edition.workspace = true
keywords.workspace = true
license.workspace = true
repository.workspace = true
rust-version.workspace = true

[features]
default = ["panic-on-logger-errors"]
alloc = []

# Generic gate to code that uses unstable features of Rust. You usually need a nightly toolchain.
unstable = []

# Helper features:
logger = []
global_allocator = []
panic_handler = []
# Ignore text output errors in logger as a workaround for firmware issues that
# were observed on the VirtualBox UEFI implementation (see uefi-rs#121).
# In those cases, this feature can be excluded by removing the default features.
panic-on-logger-errors = []
qemu = ["dep:qemu-exit", "panic_handler"] # panic_handler: logical, not technical dependency

[dependencies]
bitflags.workspace = true
log.workspace = true
ptr_meta.workspace = true
uguid.workspace = true
cfg-if = "1.0.0"
ucs2 = "0.3.2"
uefi-macros = "0.13.0"
uefi-raw = "0.5.2"
qemu-exit = { version = "3.0.2", optional = true }

[package.metadata.docs.rs]
all-features = true
rustdoc-args = ["--cfg", "docsrs"]
