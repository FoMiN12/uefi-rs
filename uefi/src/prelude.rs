//! This module is used to simplify importing the most common UEFI types.
//!
//! This includes the system table modules, `Status` codes, etc.

pub use crate::{
    boot, cstr16, cstr8, entry, runtime, system, Handle, ResultExt, Status, StatusExt,
};
