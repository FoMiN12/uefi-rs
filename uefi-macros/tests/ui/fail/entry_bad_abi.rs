#![allow(unused_imports)]
#![no_main]
#![allow(deprecated)]

use uefi::prelude::*;
use uefi_macros::entry;

#[entry]
extern "C" fn main(_handle: Handle, _st: SystemTable<Boot>) -> Status {
    Status::SUCCESS
}
