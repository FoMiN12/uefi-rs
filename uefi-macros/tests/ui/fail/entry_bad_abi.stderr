error: Entry function must have no ABI modifier
 --> tests/ui/fail/entry_bad_abi.rs:9:1
  |
9 | extern "C" fn main(_handle: Handle, _st: SystemTable<Boot>) -> Status {
  | ^^^^^^
